### DISCUSSION TITLE
Any support for virtualization tools? (Eg. Denodo)

### DISCUSSION THREAD

**Discussion by koffemaria2u** [burningalchemist/sql_exporter#564] [Q&A] [answered]
I'm wondering if there was any discussions for support for connection to virtualization tools like Denodo, which have can have multiple connections to various databases. This would allow for a single sql exporter instance to connect to the virtualization tool, instead of connecting to multiple other databases. The connection string might look something like this: 
`jdbc:<virtual_db>://<server>:<port>/<database>?user=<username>&password=<password>`

Thanks in advance and/or apologies if this has already been implemented and would appreciate redirection into the proper docs or examples.

**Comment by burningalchemist** [comment, marked as answer]
Hi @koffemaria2u, 

Apologies for the late response, it took me some time to investigate the topic.

I don't think it will be easy as there is no golang driver available out of the box. Denodo provides jdbc for JVM, and also ODBC drivers.

While we cannot do anything about java connector here (or we develop a specific driver for Denodo in Go), we can try to connect via ODBC interface.

For that you may need to:
- include odbc interface driver to sql_exporter like this one - https://github.com/alexbrainman/odbc?tab=readme-ov-file
- There is a guide on how to prepare instance (Windows/Linux) to interact with a particular ODBC driver (install unixODBC, add odbc driver, configure it, etc);

- rebuild sql_exporter using cgo feature;

- and then use 'odbc://' scheme in your connection string.

This might work (I previously tested other DB via odbc), but it definitely requires some time investment to establish/debug the connection.

I'd give it a try. 😉
